### PR TITLE
Update to the latest TypeScript version

### PR DESCRIPTION
**Pull request by adamretter**
Updates TypeScript from 4.7.4 to 5.4.5.
Closes https://github.com/evolvedbinary/jdita/pull/154, https://github.com/evolvedbinary/jdita/pull/150, https://github.com/evolvedbinary/jdita/pull/148

### CODE REVIEW

**Review by marmoure — COMMENTED** [review]
I doubled checked all the added attributes in all classes and they were good.
Just some questions and comments needs to be answered, aside from that LGTM.

File: packages/lwdita-ast/nodes/document.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import { BaseNode } from "./base";
 import { stringToChildTypes } from "@evolvedbinary/lwdita-xdita/utils";
 
+/**
+ * Interface DocumentNode defines the attribute types for a document node.
+ */
```
[the remark this answers is not in the document]
Comment (adamretter) [reply packages/lwdita-ast/nodes/document.ts:6]: Yes.

File: packages/lwdita-ast/factory.ts [hunk 2]
```diff
@@ -154,7 +154,7 @@ export function createNode(node: XMLNode<'media-source'>): MediaSourceNode;
 export function createNode(node: XMLNode<'media-track'>): MediaTrackNode;
 export function createNode(node: XMLNode<'note'>): NoteNode;
 export function createNode(node: XMLNode<'ol'>): OlNode;
-export function createNode(node: XMLNode<'p'>): PNode;
+export function createNode(node: XMLNode<'p'>): PNodeAttributes;
```
[the remark this answers is not in the document]
Comment (adamretter) [reply packages/lwdita-ast/factory.ts:157]: Same as above!

File: packages/lwdita-ast/factory.ts [hunk 3]
```diff
@@ -132,7 +132,7 @@ export function getNodeClassType(name: string): typeof BaseNode {
  */
 export function createNode(node: string): TextNode;
 export function createNode(node: XMLNode<'audio'>): AudioNode;
-export function createNode(node: XMLNode<'alt'>): AltNode;
+export function createNode(node: XMLNode<'alt'>): AltNodeAttributes;
```
[the remark this answers is not in the document]
Comment (adamretter) [reply packages/lwdita-ast/factory.ts:135]: Actually I think it is the other way around! A factory should always have the contract that it returns the interface, this then allows different implementations to be provided without breaking the API contract. I had forgotten to update all of these in `factory.ts`, I have now pushed a commit that fixes that.

**Review by marmoure — APPROVED** [review]
LGTM

File: packages/lwdita-ast/nodes/base.ts [hunk 4]
```diff
@@ -4,9 +4,99 @@ import { NonAcceptedChildError, UnknownAttributeError, WrongAttributeTypeError }
...
+   */
+  get children(): BaseNode[];
+
+  /**
+   * Getter for JSON.
+   *
+   * @privateRemarks
+   * this is not real JSON it needs to be serialized (stringified) to be converted to actual JSON
+   * 
+   * @return the JSON
```
Comment (plutonik-a) [inline packages/lwdita-ast/nodes/base.ts:31]: Typo: `@return` should be changed to `@returns`

**Review by plutonik-a — CHANGES_REQUESTED** [review]

File: packages/lwdita-ast/nodes/base.ts (hunk 4 above)
Comment (plutonik-a) [reply packages/lwdita-ast/nodes/base.ts]: Quickly fixed it with 7b5d471.

**Review by plutonik-a — APPROVED** [review]

**Pull request merged**